### DISCUSSION TITLE
PCL2 社区版介绍

### DISCUSSION THREAD

**Discussion by Pigeon0v0** [Meloong-Git/PCL#5342] [讨论]
> [!WARNING]  
> PCL2 社区版并非由上游开发者龙腾猫跃开发与管理，且与龙腾猫跃本人无关。使用 PCL2 社区版带来的一切风险请您自行承担。
> 
> PCL2 社区版下载不需要您支付任何费用。PCL-Community 组织及其成员**始终建议您使用官方版 PCL2**。
>
> 社区版出现问题请向社区版仓库反馈，不要向主线仓库反馈！

> [!IMPORTANT]
> 最新版本的社区版需要 Windows 10 20H2 或以上版本的系统以及 .NET Framework 4.8.1 才能运行。
> PCL-Community 始终建议您使用最新版本的系统和相关组件以保证系统安全和良好体验。

## 介绍

[PCL2 社区版视频介绍 - Bilibili](https://www.bilibili.com/video/BV1w5wheQEAy) | 社区版 1 群：1028074835

PCL2 Community Edition（PCL2 社区版，或 PCL2-CE，简称 CE）是由 PCL-Community 及其成员共同开发的 PCL2 下游分支，包括了主线暂未制作的功能和改进。由于个人开发水平存在较大差距，这些社区制作的功能可能存在一定瑕疵。

关于数据：社区版数据和主线不互通，独立存储于注册表中。（未来会迁移到配置文件）

关于版本号：社区版版本号并不与官方版严格对应，但会在 Release 的发布说明和启动器内关于信息中附上对应的主线版本。社区版会在主线更新时同步一次代码。

社区版于 2.10.3 (358) 引进了更新通道机制，你可以选择 Fast Ring 以获取更快速的更新，但这些版本稳定性可能较低。

欢迎大家试试看！

## 获取

你可以在 [这里](https://github.com/PCL-Community/PCL2-CE) 获取社区版源代码和发布版本。

一般用户在 Release 页面选择不带 ARM64 字样的文件下载即可。

## 差异

下表列出了社区版与官方版的一些差异。由于开发周期不定，本表可能不会及时更新。
| 项目 | 社区版 | 官方版 |
|--|--|--|
| Windows 版本要求 | Windows 10 20H2 (Build 19042) 或更高 | Windows 7 SP1 或更高 |
| .NET 版本要求 | .NET Framework 4.8.1 或以上 | .NET Framework 4.6.2 或以上 |
| Quilt 支持 | ✔️ | ❌ |
| 资源收藏夹 | ✔️ | ❌ |
| 资源包和光影包下载 | ✔️ | ❌ |
| 识别剪贴板资源链接 | ✔️ | ❌ |
| 音乐播放 SMTC 系统控制 | ✔️ 与 [Lyricify Lite](https://github.com/WXRIW/Lyricify-App) 兼容 | ❌ |
| 自动安装修改核心 | ✔️ | ❌ |
| 深色模式 | ✔️ | ❌ |
| 资源管理 | ✔️ | ⚠️ 仅文件夹快捷方式 |
| 主题 | ❌ 不会制作该功能 | ✔️ 快照版可用 |
| 回声洞 | ❌ 不会制作该功能 | ✔️ |
| 本体更新 | ⚠️ 实验性 | ✔️ |
| 启动器公告 | 🔄️ 制作中 | ✔️ |
| 开发和更新周期 | 较快，但不固定 | 平均每月一次 |
| 开发者 | 社区开发者 | 龙腾猫跃 |

**Comment by LinQingYuu** [comment]
补充：因为成本问题，社区版没有数字签名（随便一张都是 2k+ 起步的）

~Edit:等下好像可以用自签名证书~

**Reply by WhatDamon** [reply]
自签名有效期相对而言较短，即便也不是不能用，但可能会增加被误报的风险（尤其是国内那几个杀毒软件自签名相比不签名更容易被杀）
如果实在需要，应该可以从Certum花小钱申请带`Open Source`前缀的开源的数字签名（应该是适用于像PCL这样的半开源软件，称其半主要是因为许可证并不符合开源规范，望理解）

**Reply by LinQingYuu** [reply]
>自签名有效期相对而言较短

自签名的证书有效期可以无限长，因为你签发证书的时候可以设定过期时间，比如一张 1000 年的证书，主要是加个证书信任应该 ~没问题~ 会被用户加以揍死（主要是看杀软会不会处理这种情况）

**Reply by allMagicNB** [reply]
所以有计划吗？

**Comment by Pigeon0v0** [comment]
社区版计划于 2.9.4 或 2.10.0 版本放弃对 Windows 10 1607 以下版本系统的支持，因为使用到了高版本系统的特性。

如果有用社区版的可以吱一声，看看大家的意见？

**Reply by LinQingYuu** [reply]
个人意见:我无所谓，手上系统最旧的都 Win 10 1703 了 ~（什么世纪老古董）~

**Reply by allMagicNB** [reply]
除了社区成员应该基本没人用（我也不用）

**Reply by CreeperIsASpy** [reply]
我们学校机房电脑都win10了.png

**Reply by allMagicNB** [reply]
鉴定为：这会进一步减小使用人群范围。

**Reply by LinQingYuu** [reply]
> 我们学校机房电脑都win10了.png

我们 Win 11（

**Reply by LinQingYuu** [reply]
>鉴定为：这会进一步减小使用人群范围。

![StatCounter-windows_version-ww-monthly-202312-202412](https://github.com/user-attachments/assets/702771e4-8d6d-43aa-996f-c4e096c89653)

根据 StatCounter 的统计数据，Win 7 占比低于 2.5%，排除掉政企机关及 ATM/银行金融部门，个人预计可能占比不到 1.5%

**Reply by 3gf8jv4dv** [reply]
我个人持支持态度。

**Reply by allMagicNB** [reply]
> 
> >鉴定为：这会进一步减小使用人群范围。
> 
> ![StatCounter-windows_version-ww-monthly-202312-202412](https://github.com/user-attachments/assets/702771e4-8d6d-43aa-996f-c4e096c89653)
> 
> 根据 StatCounter 的统计数据，Win 7 占比低于 2.5%，排除掉政企机关及 ATM/银行金融部门，个人预计可能占比不到 1.5%

☝🤓你没算 Windows 8。

**Reply by SALTWOOD** [reply]
提问：是什么特性非要 Windows 10 及以上版本？

**Reply by 3gf8jv4dv** [reply]
> 提问：是什么特性非要 Windows 10 及以上版本？

https://github.com/PCL-Community/PCL2-CE/pull/48

**Reply by LinQingYuu** [reply]
> ☝🤓你没算 Windows 8。

8 和 8.1 的总量更少了，加起来 2.18 %，考虑到其他因素（比如不玩 Minecraft），还是低于 2%

**Reply by WorldHim** [reply]
> 鉴定为：这会进一步减小使用人群范围。

这本身也并非是适合所有人使用的版本

**Reply by Deep-Dark-Forest** [reply]
> 鉴定为：这会进一步减小使用人群范围。

这么说吧，使用 Windows 10 1607 以下版本系统的人很少，这些人里知道/使用 PCL 的人更少，知道/使用 PCL 的人里知道/使用 PCL 社区版的人就几乎没有了……所以这个影响微乎其微。而且那么多知名软件（Edge、Steam 等等）都已经停止了对 Win10 以下的支持

**Reply by xiaoniaoawa** [reply]
对的，实在不理解那些连QQ微信都用不了的系统怎么还有人在拿它打游戏...

**Comment by MFn233** [comment]
真心希望有好的特性龙猫可以merge啊

**Reply by LinQingYuu** [reply]
>真心希望有好的特性龙猫可以merge啊

一般情况下这种东西会提 PR 的（如果真做了的话）

-----
[至少在这一刻 热爱不问为何 存在为将心声响彻](https://mzh.moegirl.org.cn/%E5%85%89%E4%B8%8E%E5%BD%B1%E7%9A%84%E5%AF%B9%E7%99%BD)

**Comment by Deep-Dark-Forest** [comment]
@Pigeon0v0 要不在表格里加上官方版即将不可使用 MCIM 镜像源？
https://github.com/Hex-Dragon/PCL2/issues/5421#issuecomment-2600938152

**Reply by Pigeon0v0** [reply]
这个修改我不确定会不会同步到 CE...

**Reply by Deep-Dark-Forest** [reply]
> 这个修改我不确定会不会同步到 CE...

那就把 CE 改成可以让用户自行选择是否使用 MCIM 吧（

**Reply by z0z0r4** [reply]
> 这个修改我不确定会不会同步到 CE...

话说现在的情况是什么，我才看到还有这个discussion

**Reply by Pigeon0v0** [reply]
> > 这个修改我不确定会不会同步到 CE...
> 
> 话说现在的情况是什么，我才看到还有这个discussion

现在貌似是在某次主线更新的时候给合掉了...（

需要的话可以加回去

**Reply by z0z0r4** [reply]
> > > 这个修改我不确定会不会同步到 CE...
> > 
> > 话说现在的情况是什么，我才看到还有这个discussion
> 
> 现在貌似是在某次主线更新的时候给合掉了...（
> 
> 需要的话可以加回去

我个人欢迎接入，没有接入直接导致新mod采集延缓，尤其是 curseforge 可以说是停滞

需要的话加回去这种操作其实有点抽象了。也许可以单独开个 issue 跟进讨论之类
~~评估可靠性~~

**Comment by WhatDamon** [comment]
我有以下建议，由于内容过于繁杂现在这里提出，**_可以用的一些建议接纳以后将会添加到Issues内_**
## 1. .NET Framework to .NET
既然选择了这个项目，我是不太相信社区开发者还搞不清楚.NET和.NET Framework
可以尝试将项目**升级至.NET 6或更高版本**（需要使用`dotnet`的`update-assistant`或者VS插件），当然更需要手动更新代码风格不然升级完会报错
这样可以精简掉部分NuGet包，例如`Costura`（新的.NET原生支持）、`Newtonsoft.Json`（使用自带的`System.Text.Json`替代）、`Ookii.Dialogs.Wpf`（诸如TaskDialog目前.NET有更简单的写法了，但是覆盖比较有限）
此外.NET相比.NET Framework的开源特性也可以让整个项目在一定程度上更加公开透明，跨平台特性如果能迁移至Avalonia（本项目使用WPF，迁移难度应该不会特别高）那就能允许其他操作系统的用户使用PCL2
最大的缺点就是需要用户自行下载运行库，不过.NET 8本身就需要Windows 10（哪怕Windows 7配置后也可以跑，会有爆内存风险，.NET 6是最后一个原生支持Windows 7的新.NET版本）

## 2. 关于注册表存储配置
对于注册表存储配置个人更建议使用单独的配置文件存储在文件根目录，让软件更加绿色化（减少痕迹保留，使应用程序更加便携化，注册表除非引入PortableApp否则没法做到不同设备配置同步即插即用），实质上作为一个相比官方启动器对系统侵入性更低的启动器程序而言，没必要为了炫技或者意义不是很大的不兼容原版PCL2而选择反其道而行之（说到这里其实可以添加导入原版配置的功能）

## 3. 前期开发管理
该项目疑似存在多余的NuGet包，建议前期以代码改进为主，优化屎山，而不是一味增加新功能降低代码可维护性
目前我已经在项目中发现了部分不应该存在于社区版的东西并未处理干净，引导用户安装的还是原版要求的.NET Framework 4.6.1(见`Application.xaml.vb`)
还有由于技术债，项目的部分功能引用的函数方式已经过时，建议更新

## 4. 可能的NativeAOT
虽然对于一个（半）开源项目而言必要性不高，但是这应该是有实际可操作性的

---

不过很多功能我都没法去做，因为我对C#更加熟悉，对VB.NET了解甚少（即便作为.NET语言两者有共通之处），除非项目整体迁移至C#
其中部分建议也会增加同步上游的难度，请酌情解决

**Reply by Pigeon0v0** [reply]
1. 之前尝试过从 .NET Framework 迁移到 .NET (Core)，但是由于很多语法上都不太一样，迁移工作量是真不小...有时间再看看吧
2. 会考虑的，但由于注册表配置不少都是设备端加密的，估计做成 Portable 意义也不大...
3. 目前的代码改进基本都是做到哪里发现不太行再改到哪里，从头翻一遍也要不少时间...
4. 等 1 完成再看看？

**Reply by LinQingYuu** [reply]
>（说到这里其实可以添加导入原版配置的功能）

官方版部分内容是加密的，两端用的识别码生成逻辑不太一样，这个没办法

>Newtonsoft.Json（使用自带的System.Text.Json替代）

就算真干到 .NET 8.0 Newtosoft.Json 也不会那么快换掉

**Reply by whitecat346** [reply]
> 目前的代码改进基本都是做到哪里发现不太行再改到哪里，从头翻一遍也要不少时间...

对于这个还是建议抽出时间进行优化的，毕竟PCL2也是个老东西了，屎山啥的还是有很多，进行优化对后面的发展有很大的方便的

**Reply by LinQingYuu** [reply]
>原版要求的.NET Framework 4.6.1(见Application.xaml.vb)

Fix at https://github.com/PCL-Community/PCL2-CE/pull/69

**Reply by Hill23333** [reply]
关于您说的第一点建议我有以下见解：
关于迁移 .NET Framework 的事情，其实社区内已经尝试过，但是失败了（详情看[这个项目](https://github.com/PCL-Community/PCL2-Next)）。另外迁移到 Avalonia 其实[已经在制作](https://github.com/PCL-Community/PCL2.Neo)，但是并不是`迁移难度应该不会特别高`。实际上，迁移到 Avalonia 几乎要重写所有代码。不过借此机会也可以完全重构屎山。另外其实 PCLCE 已经不支持 Windows 7，必须 Windows 10 20H2+  的系统了，所以将来也是会迁移 .NET8+（如果迁移的话）

另外：**WPF 不支持 Native AOT**（因为反射），别想了。

**Comment by LingXi9374** [comment]
如果开发者有移植Linux的打算，考虑一下使用Avalonia框架，是一个跨平台的框架，API基本与.NET Framework差不多，但总体来看移植难度仍然很大，比如Page和Frame控件的实现，但有个文档我个人认为已经解决了一些难点

文档链接：https://amwx.github.io/FluentAvaloniaDocs/pages/Controls/Frame

如果开发者有能力支持Linux版本的开发，我会表示支持，如果没有办法成功实现我也表示理解，最后我也会点个Star支持一下的(=^▽^=)

**Reply by LinQingYuu** [reply]
抱歉，社区暂时没有这个计划，这需要等到社区版技术栈迭代到可以跨平台运行的 .NET Core

**Reply by Hill23333** [reply]
目前基于 Avalonia 的 [PCL2.Neo](https://github.com/PCL-Community/PCL2.Neo) 正在制作，同时社区内也使用其他框架尝试跨平台（如使用 Tauri 的 [PCL II Nova](https://github.com/PCL-Community/PCL2.Nova.App)）

如果可以的话可以向上面两个项目贡献代码！

**Comment by lingyunmo** [comment]
很好的版本，使我旋转跳跃

**Comment by Harry-hq** [comment]
移植linux我认为可以通过wine，ce版加一个在wine下使用wineconsole运行linux下的java的功能，可以省不少资源
把PCL2和wine以及.NET Framework 4.6.2打包到一起，放到apt上（doge）

**Reply by wuliaodexiaoluo** [reply]
>移植linux我认为可以通过wine，ce版加一个在wine下使用wineconsole运行linux下的java的功能，可以省不少资源
把PCL2和wine以及.NET Framework 4.6.2打包到一起，放到apt上（doge）

.NET Framework 不开源，公共分发需要 Microsoft 许可，Ubuntu/Debian 官方源大概率不会接受这个

**Comment by ZEleven10** [comment]
no

________________________________
发件人: allMagic ***@***.***>
发送时间: 2025年5月2日 7:12
收件人: Hex-Dragon/PCL2 ***@***.***>
抄送: ZEleven10 ***@***.***>; Manual ***@***.***>
主题: Re: [Hex-Dragon/PCL2] PCL2 社区版介绍 (Discussion #5342)


所以有计划吗？

―
